src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -95,8 +95,20 @@ int MPIR_TSP_sched_new_type(MPIR_TSP_sched_t s, MPIR_TSP_sched_issue_fn issue_fn
                             MPIR_TSP_sched_complete_fn complete_fn, MPIR_TSP_sched_free_fn free_fn)
 {
     MPII_Genutil_sched_t *sched = s;
-    MPII_Genutil_vtx_type_t *newtype;
-    int type_id;
+    MPII_Genutil_vtx_type_t *vtype, *newtype;
+    int type_id, i;
+
+    /* only free_fn can be NULL */
+    MPIR_Assert(issue_fn && complete_fn);
```

Comment:
this seems unrelated to this commit

Reply:
Sure, should I make a new commit for that?

Reply:
sure

Reply:
done